homeassistant/components/ollama/config_flow.py

```diff
@@ -0,0 +1,148 @@
+"""Config flow for Ollama integration."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+import sys
+from types import MappingProxyType
+from typing import Any
+
+import httpx
+import ollama
+import voluptuous as vol
+
+from homeassistant.config_entries import (
+    ConfigEntry,
+    ConfigFlow,
+    ConfigFlowResult,
+    OptionsFlow,
+)
+from homeassistant.const import CONF_URL
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import config_validation as cv
+from homeassistant.helpers.selector import (
+    NumberSelector,
+    NumberSelectorConfig,
+    NumberSelectorMode,
+    TemplateSelector,
+    TextSelector,
+    TextSelectorConfig,
+    TextSelectorType,
+)
+
+from .const import (
+    CONF_MAX_HISTORY,
+    CONF_MODEL,
+    CONF_PROMPT,
+    DEFAULT_PROMPT,
+    DEFAULT_TIMEOUT,
+    DOMAIN,
+    MAX_HISTORY_NO_LIMIT,
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+STEP_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_URL): TextSelector(
+            TextSelectorConfig(type=TextSelectorType.URL)
+        ),
+        vol.Required(CONF_MODEL): cv.string,
+    }
+)
+
+
+async def validate_input(hass: HomeAssistant, data: dict[str, Any]) -> None:
+    """Validate the user input allows us to connect.
+
+    Data has the keys from STEP_USER_DATA_SCHEMA with values provided by the user.
+    """
+    client = ollama.AsyncClient(host=data[CONF_URL])
+    async with asyncio.timeout(DEFAULT_TIMEOUT):
+        await client.list()
```

Comment:
doesn't this return the list of models?

Reply:
Yes, but only the ones that are already downloaded